torch/_inductor/compile_fx.py

```diff
@@ -296,11 +298,29 @@ def compile_fx_inner(
             if isinstance(t, torch.Tensor)
         )
 
+        def _has_dynamism(t):
+            if isinstance(t, torch.Tensor):
+                if not all(isinstance(s, (int, sympy.Integer)) for s in t.size()):
+                    return True
+                if not all(isinstance(s, (int, sympy.Integer)) for s in t.stride()):
+                    return True
+            elif isinstance(t, sympy.Symbol):
+                if isinstance(t, sympy.Integer):
+                    return False
+                return True
+            return False
```

Comment:
Instead of this use free_symbols from symbolic_shapes

Reply:
No, I mean, use this:

```
def free_symbols(val: Union[SymInt, torch.Tensor]) -> Set[sympy.Symbol]:
    if isinstance(val, (SymInt, SymFloat)):
        return val.node.expr.free_symbols
    elif isinstance(val, (int, float, bool)):
        return set()
    elif isinstance(val, torch.Tensor):
        r = set()
        for s in val.size():
            r |= free_symbols(s)
        for s in val.stride():
            r |= free_symbols(s)
        r |= free_symbols(val.storage_offset())
        return r
    else:
        raise AssertionError(f"cannot compute free_symbols of {val}")
```

in torch.fx.experimental.symbolic_shapes

Reply:
Oh very nice.